src/App.tsx

```diff
@@ -15,48 +24,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(
+  goods: string[],
+  sortFiled: SortType,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortFiled) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortFiled) {
+        case SortType.Alphabet:
+          return goodA.localeCompare(goodB);
+
+        case SortType.Length:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortFiled, setSortFiled] = useState<SortType>(SortType.Empty);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortFiled,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortFiled !== SortType.Alphabet },
+          )}
+          onClick={() => setSortFiled(SortType.Alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortFiled !== SortType.Length },
+          )}
+          onClick={() => setSortFiled(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': isReversed === false },
+          )}
+          onClick={
+            isReversed === false
+              ? () => setIsReversed(true)
+              : () => setIsReversed(false)
+          }
```

Comment:
It is incorrect
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state